darwin/exporter/formats/coco.py

```diff
@@ -3,33 +3,39 @@
 from pathlib import Path
 from typing import Any, Dict, Iterator, List, Optional
 
+import deprecation
 import numpy as np
 from upolygon import draw_polygon, rle_encode
 
 import darwin.datatypes as dt
+from darwin.exporter.formats.numpy_encoder import NumpyEncoder
 from darwin.utils import convert_polygons_to_sequences
-
-
-class NumpyEncoder(json.JSONEncoder):
-    def default(self, obj: Any) -> Any:
-        if isinstance(obj, np.integer):
-            return int(obj)
-        elif isinstance(obj, np.floating):
-            return float(obj)
-        elif isinstance(obj, np.ndarray):
-            return obj.tolist()
-        else:
-            return super(NumpyEncoder, self).default(obj)
+from darwin.version import __version__
 
 
 def export(annotation_files: Iterator[dt.AnnotationFile], output_dir: Path) -> None:
-    output = build_json(list(annotation_files))
-    # TODO, maybe an optional output name (like the dataset name if available)
+    """
+    Exports the given ``AnnotationFile``s into the coco format inside of the given ``output_dir``.
+
+    Parameters
+    ----------
+    annotation_files : Iterator[dt.AnnotationFile]
+        The ``AnnotationFile``s to be exported.
+    output_dir : Path
+        The folder where the new coco file will be.
+    """
+    output = _build_json(list(annotation_files))
     output_file_path = (output_dir / "output").with_suffix(".json")
     with open(output_file_path, "w") as f:
         json.dump(output, f, cls=NumpyEncoder, indent=1)
 
 
+@deprecation.deprecated(
+    deprecated_in="0.7.7",
+    removed_in="0.8.0",
+    current_version=__version__,
+    details="Use the ``export`` instead.",
```

Comment:
I'm not sure this `details` string is very well descriptive of the reason why we're deprecating these functions. The following is an attempt at making it a bit better, but lmk what you think
```suggestion
    details="This function is going to be turned into private. This means that breaking changes in its interface and implementation are to be expected. We encourage using ``export`` instead of calling this low-level function directly.",
```

Reply:
I like the idea!
I was following the convention the library suggests in the docs, but I think your description makes more sense!